src/hammer-vlsi/hammer_vlsi/constraints.py

```diff
@@ -242,42 +244,158 @@ class PlacementConstraint(NamedTuple('PlacementConstraint', [
     __slots__ = ()
 
     @staticmethod
-    def from_dict(constraint: dict) -> "PlacementConstraint":
+    def from_dict(masters: List[MacroSize], constraint: dict) -> "PlacementConstraint":
```

Comment:
Hmm, could we make the `masters` parsing into a separate function? Namely I'd like all Hammer objects to be serializable and deserializable from dict<->Python. Maybe it could be a second optional argument or second override or something. As for right now this will break some of the Hammer IR stuff.

Reply:
Hmm that's kind of a major rewrite, but let me see what I can do. Basically, I would create a new function that has these same args, and do the `master` resolving there, then pass that to the `from_dict` method which would then make `width` and `height` mandatory again.

Reply:
Sounds good to me.

Reply:
Resolved, please take a look.

Reply:
Looks good, thanks!